validation/README.md

```diff
@@ -0,0 +1,112 @@
+
+1. Sample IDs
+    - Within metamist, samples are identified by both the external and internal IDs. Within the MT dump, the samples are referred to by their External IDs, whilst query operations on metamist objects are based on the CPG IDs. The first Metamist query is to pull a mapping of the Internal:External IDs through the `get_external_participant_id_to_internal_sample_id` method. Using this we can swap between IDs interchangeably.
+2. Analysis truth
+    - When running the variant data validation we need to locate the reference truth data; both the expected variant calls, and region file representing confidently called regions. In this context, the 'truth' data is treated as an Analysis result (it was not analysed by the CPG, but is an analysis product). Analysis entries have been created to store this information - see below.
+
+#### Analysis objects
+
+The Analysis entries in Metamist are relevant at 2 points:
+
+1. Sourcing the data to use as a base truth
+2. Recording the results of the analysis
+
+Truth data is in the form of an expected set of variant calls, and a BED file. As well as being able to pull this data, we also need the ability to version the truth if the call-set develops over time. Metamist provides for both these requirements:
+
+1. Analysis objects are atomic, and adding a new analysis object does not replace a previous one (though this is possible through patching); this means we can have multiple 'truth' analyses per sample at any one time
+2. Analyses have a Type from a strict enumeration, so we can store these as `custom` analyses, and they will not clash with main-pipeline records
+    -we could extend this enumeration to include a specific `validation` type if required
+3. Analyses have an attached `Active` Boolean flag - if a truth set is replaced it would be set to inactive rather than deleted
+
+The `meta` attribute in the Analysis objects permits storage of arbitrary data, so we can store any objects relevant to the anaylsis without creating more granular objects.
+
+An Analysis entry has been created for each sample, with `output` indicating the analysis product (the truth VCF) and `analysis.meta` containing a pointer to the confident region BED file to use:
+
+```python
+from sample_metadata.apis import AnalysisApi
+from sample_metadata.model.analysis_model import AnalysisModel
+from sample_metadata.model.analysis_type import AnalysisType
+from sample_metadata.model.analysis_status import AnalysisStatus
+
+anal_api = AnalysisApi()
+anal_api.create_new_analysis(
+    project='validation',
+    analysis_model=AnalysisModel(
+        sample_ids=['CPG_ID', 'CPG_ID2'],
+        type=AnalysisType('custom'),
+        status=AnalysisStatus('completed'),
+        output='gs://cpg-validation-test/HG001/HG001_GRCh38_1_22_v4.2.1_benchmark.vcf.gz',
+        meta={
+            'confident_region': 'gs://cpg-validation-test/HG001/HG001_GRCh38_1_22_v4.2.1_benchmark.bed',
```

Comment:
As you mentioned, we'd need either a `validation` analysis type, or some flag in meta to allow distinguishing validation analysis entries from others, when we need to query them later:

```
a_query_model = AnalysisQueryModel(
        projects=[get_config()['workflow']['dataset']],
        sample_ids=[cpg_id],
        type=AnalysisType('custom'),
        active=True,
        meta={'type': 'validation'}
    )
```

Reply:
Ah dang, the README is now going to be hopelessly out of data. I'll blast through that now